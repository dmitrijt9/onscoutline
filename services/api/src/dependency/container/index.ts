import { Connection, getCustomRepository } from 'typeorm'
import { CompetitionRepository } from '../../repositories/CompetitionRepository'
import { FacrScraper } from '../../services/scrapers/FacrScraper'
import { Scraper } from '../../services/scrapers/Scraper'
import { AppConfig, getAppConfig } from '../config/index'
import { bootstrapDbConnection } from './bootstrap/db-connection'

export const createContainer = async (
    appConfig = getAppConfig(process.env),
): Promise<Container> => {
    const typeormConnection = await bootstrapDbConnection(appConfig)

    const competitionRepository = getCustomRepository(CompetitionRepository)

    const scraper = new Scraper()
    const facrScraper = new FacrScraper(scraper, appConfig, competitionRepository)

    return {
        config: appConfig,
        typeormConnection,
        competitionRepository,
        facrScraper: facrScraper,
    }
}

export interface Container {
    config: AppConfig
    typeormConnection: Connection
    competitionRepository: CompetitionRepository
    facrScraper: FacrScraper
}
