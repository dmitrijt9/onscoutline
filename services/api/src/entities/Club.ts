import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm'

@Entity()
export class Club {
    @PrimaryGeneratedColumn()
    id: number

    @Column('varchar', { unique: true, nullable: true })
    facrId?: string

    @Column()
    name: string

    @Column('varchar', { nullable: true })
    facrUuid?: string
}
