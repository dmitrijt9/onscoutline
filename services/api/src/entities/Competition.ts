import { Column, Entity, Index, PrimaryGeneratedColumn } from 'typeorm'

@Entity()
@Index('competition_facr_id', ['regionId', 'facrId'])
export class Competition {
    @PrimaryGeneratedColumn()
    id: number

    @Column()
    name: string

    @Column()
    regionId: string

    @Column()
    regionName: string

    @Column()
    facrId: string

    @Column()
    facrUuid: string
}
