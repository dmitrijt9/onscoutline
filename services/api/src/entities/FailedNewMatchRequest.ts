import { NewMatchRequest } from '../services/match/types'
import { jsonTransformer } from '../utils/typeorm/jsonTransformer'
import { Column, CreateDateColumn, Entity, PrimaryGeneratedColumn, UpdateDateColumn } from 'typeorm'

@Entity({})
export class FailedNewMatchRequest {
    @PrimaryGeneratedColumn('uuid')
    id: string

    @Column()
    matchFacrUuid: string

    @Column('varchar')
    status: FailedNewMatchRequestStatus

    @Column('longtext', { transformer: jsonTransformer('requestJson') })
    requestJson: NewMatchRequest

    @CreateDateColumn()
    createdAt?: string

    @UpdateDateColumn()
    updatedAt?: string
}

export enum FailedNewMatchRequestStatus {
    Unprocessed = 'Unprocessed',
    UnprocessedClub = 'UnprocessedClub',
    UnprocessedPlayer = 'UnprocessedPlayer',
    Processed = 'Processed',
}
