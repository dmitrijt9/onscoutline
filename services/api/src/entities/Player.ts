import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm'

@Entity()
export class Player {
    @PrimaryGeneratedColumn()
    id: number

    @Column()
    name: string

    @Column()
    surname: string

    @Column('date')
    dateOfBirth: string

    @Column('date')
    facrMemberFrom: string
}
