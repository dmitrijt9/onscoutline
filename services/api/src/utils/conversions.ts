// Convert date to YYYY-MM-DD string
export const toOnscoutlineDateFormat = (date?: Date) => {
    const now = date ?? new Date()

    return `${now.getFullYear()}-${(now.getMonth() + 1).toString().padStart(2, '0')}-${now
        .getDate()
        .toString()
        .padStart(2, '0')}`
}
