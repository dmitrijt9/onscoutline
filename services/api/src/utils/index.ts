export const sleep = async (time: number) => {
    await new Promise((resolve) => {
        setTimeout(resolve, time)
    })
}

export const isNil = <T>(value: T | null | undefined): value is null | undefined =>
    value === null || value === undefined
