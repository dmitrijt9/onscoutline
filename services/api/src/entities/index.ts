import { Club } from './Club'
import { Competition } from './Competition'
import { Match } from './Match'
import { Player } from './Player'
import { CompetitionHasSeason } from './Relations/CompetitionHasSeason'
import { PlayerInClub } from './Relations/PlayerInClub'
import { PlayerInMatch } from './Relations/PlayerInMatch'
import { Season } from './Season'

export const entities = [
    Club,
    Competition,
    Player,
    PlayerInClub,
    Match,
    PlayerInMatch,
    Season,
    CompetitionHasSeason,
]
