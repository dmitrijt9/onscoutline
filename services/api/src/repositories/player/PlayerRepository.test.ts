import { Container } from '../../dependency/container/index'
import {
    cleanDb,
    createTestingClient,
    stopTestApplication,
    TestingClient,
} from '../../dependency/test-utils/index'
import { Gender } from '../../entities/Player'
import { toOnscoutlineDateFormat } from '../../utils/conversions'

describe('Player Repository', () => {
    let testingClient: TestingClient
    let container: Container

    beforeAll(async () => {
        testingClient = await createTestingClient()
        container = testingClient.container
    })

    afterAll(async () => {
        await stopTestApplication(testingClient.application)
    })

    beforeEach(async () => {
        await cleanDb(testingClient)
    })

    it('should save a player without problems', async () => {
        const { playerRepository } = container

        const savedPlayer = await playerRepository.save({
            facrId: '123457',
            facrMemberFrom: toOnscoutlineDateFormat(new Date('2005-03-16')),
            name: 'Kristian',
            surname: 'Ronaldny',
            gender: Gender.Male,
            country: 'Portugalsko',
            dateOfBirth: toOnscoutlineDateFormat(new Date('1990-03-16')),
        })

        expect(savedPlayer).not.toBeNull()
        const { id, ...savedPlayerWithoutId } = savedPlayer
        expect(savedPlayerWithoutId).toMatchSnapshot()
    })

    it('should find a player by his fullname', async () => {
        const { playerRepository } = container

        const savedPlayer1 = await playerRepository.save({
            facrId: 'prvni',
            facrMemberFrom: toOnscoutlineDateFormat(new Date('2005-03-16')),
            name: 'Kristian',
            surname: 'Ronaldny',
            gender: Gender.Male,
            country: 'Portugalsko',
            dateOfBirth: toOnscoutlineDateFormat(new Date('1990-03-16')),
        })

        const savedPlayer2 = await playerRepository.save({
            facrId: 'druhej',
            facrMemberFrom: toOnscoutlineDateFormat(new Date('2005-03-16')),
            name: 'Javier Alando',
            surname: 'Martinez Suarez',
            gender: Gender.Male,
            country: 'Portugalsko',
            dateOfBirth: toOnscoutlineDateFormat(new Date('1990-03-16')),
        })

        const foundPlayer1 = await playerRepository.findByFullname('Ronaldny Kristian')
        const foundPlayer2 = await playerRepository.findByFullname('Martinez Suarez Javier Alando')

        expect(foundPlayer1).not.toBeNull()
        expect(foundPlayer2).not.toBeNull()

        expect(foundPlayer1).toEqual(savedPlayer1)
        expect(foundPlayer2).toEqual(savedPlayer2)
    })
})
