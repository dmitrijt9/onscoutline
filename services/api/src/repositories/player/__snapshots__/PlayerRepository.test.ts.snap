// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player Repository should save a player without problems 1`] = `
Object {
  "country": "Portugalsko",
  "dateOfBirth": "1990-03-16",
  "facrId": "123457",
  "facrMemberFrom": "2005-03-16",
  "gender": "Male",
  "name": "Kristian",
  "positions": null,
  "shirtNumber": null,
  "surname": "Ronaldny",
  "transferRecords": null,
}
`;
